scripts/template_openxr_dispatchdynamic.hpp

```diff
@@ -59,30 +56,52 @@ class DispatchLoaderDynamic {
      * @name Constuctor/Factory functions
      * @{
      */
+/*{protect_proto_begin()}*/
     /*!
      * @brief Create a lazy-populating dispatch table.
      *
      * If getInstanceProcAddr is not supplied, the static ::xrGetInstanceProcAddr will be used.
      */
-    explicit DispatchLoaderDynamic(Instance instance = nullptr, PFN_xrGetInstanceProcAddr getInstanceProcAddr = nullptr)
+    explicit DispatchLoaderDynamic(Instance instance = nullptr, PFN_xrGetInstanceProcAddr getInstanceProcAddr = nullptr) noexcept
         : m_instance(instance), pfnGetInstanceProcAddr(reinterpret_cast<PFN_xrVoidFunction>(getInstanceProcAddr)) {
         if (pfnGetInstanceProcAddr == nullptr) {
             pfnGetInstanceProcAddr = reinterpret_cast<PFN_xrVoidFunction>(&::xrGetInstanceProcAddr);
         }
     }
+#else
+    /*!
+     * @brief Create a lazy-populating dispatch table.
+     */
+    DispatchLoaderDynamic() noexcept = default;
 
     /*!
-     * @brief Create a fully-populated dispatch table given a non-null XrInstance and an optional getInstanceProcAddr.
-     *
-     * If getInstanceProcAddr is not supplied, the static ::xrGetInstanceProcAddr will be used.
+     * @brief Create a lazy-populating dispatch table and populates the functions that do not need an instance.
+     */
+    DispatchLoaderDynamic(PFN_xrGetInstanceProcAddr getInstanceProcAddr) noexcept {
+        populateBase(getInstanceProcAddr);
+    }
+
+    /*!
+     * @brief Populates the functions that do not need an instance.
+     */
+    void populateBase(PFN_xrGetInstanceProcAddr getInstanceProcAddr) noexcept {
```

Comment:
I'd actually like to allow this to be used in all cases, not just when the no-proto thing is defined.